source/core/box.ts

```diff
@@ -0,0 +1,225 @@
+        return new Float32Array([
+            0.0, 0.0,
+            1.0, 0.0,
+            0.0, 1.0,
+
+            0.0, 1.0,
+            1.0, 0.0,
+            1.0, 1.0,
+
+            0.0, 0.0,
+            1.0, 0.0,
+            0.0, 1.0,
+
+            0.0, 1.0,
+            1.0, 0.0,
+            1.0, 1.0,
+
+            0.0, 0.0,
+            1.0, 0.0,
+            0.0, 1.0,
+
+            0.0, 1.0,
+            1.0, 0.0,
+            1.0, 1.0,
+
+            0.0, 0.0,
+            1.0, 0.0,
+            0.0, 1.0,
+
+            0.0, 1.0,
+            1.0, 0.0,
+            1.0, 1.0,
+
+            0.0, 0.0,
+            1.0, 0.0,
+            0.0, 1.0,
+
+            0.0, 1.0,
+            1.0, 0.0,
+            1.0, 1.0,
+
+            0.0, 0.0,
+            1.0, 0.0,
+            0.0, 1.0,
+
+            0.0, 1.0,
+            1.0, 0.0,
+            1.0, 1.0,
+        ]);
+    }
+
+    /**
+     * Binds the vertex buffer object (VBO) to an attribute binding point of a given, pre-defined index.
+     */
+    protected bindBuffers(indices: Array<GLuint>): void {
+        /* Please note the implicit bind in attribEnable. */
+        this._buffers[0].attribEnable(0, 3, this.context.gl.FLOAT, false, 0, 0, true, false);
+        if (this._textured) {
+            this._buffers[1].attribEnable(1, 2, this.context.gl.FLOAT, false, 0, 0, true, false);
+        }
+    }
+
+    /**
+     * Unbinds the vertex buffer object (VBO) and disables the binding point.
+     */
+    protected unbindBuffers(indices: Array<GLuint>): void {
+        /* Please note the implicit unbind in attribEnable is skipped. */
+        this._buffers[0].attribDisable(0, true, true);
+        if (this._textured) {
+            this._buffers[1].attribDisable(1, true, true);
+        }
+    }
+
+    /**
+     * Creates the vertex buffer object (VBO) and creates and initializes the buffer's data store.
+     * @param aVertex - Attribute binding point for vertices.
+     */
+    initialize(aVertex: GLuint, aTexCoord: GLuint): boolean {
+        const gl = this.context.gl;
+
+        const valid = super.initialize([gl.ARRAY_BUFFER, gl.ARRAY_BUFFER], [aVertex, aTexCoord]);
+
+        assert(this._buffers[0] !== undefined && this._buffers[0].object instanceof WebGLBuffer,
+            `expected valid WebGLBuffer`);
+
+        if (this._textured) {
+            assert(
+                this._buffers[1] !== undefined && this._buffers[1].object instanceof WebGLBuffer,
+                `expected valid WebGLBuffer`);
+        }
+
+        this._buffers[0].data(this.generateVertices(this._width, this._height, this._depth), gl.STATIC_DRAW);
+        if (this._textured) {
+            this._buffers[1].data(this.generateTexCoords(), gl.STATIC_DRAW);
+        }
+
+        return valid;
+    }
+
+    /**
+     * Draws the box.
+     */
+    draw(): void {
+        const gl = this.context.gl;
+        gl.drawArrays(gl.TRIANGLES, 0, 36);
+    }
+
```

Comment:
expose texCoordLocation and vertexLocation via getter - this is a new concept already applied in another library, but not  done in webgl-operate yet:
```
    /**
     * Attribute location to that this geometry's vertices are bound to.
     */
    get vertexLocation(): GLint {
        return this._vertexLocation;
    }

    /**
     * Attribute location to that this geometry's texture coordinates are bound to.
     */
    get texCoordLocation(): GLint {
        return this._texCoordLocation;
    }

    /**
     * Attribute location to that this geometry's id data is bound to.
     */
    get idLocation(): GLint {
        return this._idLocation;
    }
...
```